webui: switch from  mod_php to php-fpm (#1287)
bareos/bareos: docs/manuals/source/IntroductionAndTutorial/InstallingBareosWebui.rst

-  The |webui| can be installed on any host. It does not have to be installed on the same as the |dir|.

> -  PHP-FPM with Apache HTTP webserver having mod-rewrite enabled.

Reviewer: ```suggestion
-  The default installation uses PHP-FPM with Apache HTTP webserver having mod-rewrite enabled.
```

Reviewer: How about upgrading from bareos-21 with mod_php to bareos-22 with php-fpm? 
What needs to be done here? 
Has that been tested?
We should definitely have a documentation what needs to be done AND a breaking changes info in CHANGELOG.md

Author: I agree, we should provide a section `Upgrade from 21.x to 22.x` in our documentation in the `Installing Bareos Webui` chapter. Currently testing upgrades and gathering information, will push a commit on top when done.

Author: Applied, see https://github.com/bareos/bareos/pull/1287/commits/da48955442b440c0892f8862c65fa795c4a23872 .

   We do provide a specific JobId in the *JobId* command argument in this example. Only the *JobId* given by the placeholder %i will be computed into the cache.

Upgrade from Bareos 21 or lower to Bareos 22
--------------------------------------------

|webui| now requires **php-fpm** instead of Apache **mod-php**. Due to this change there are some manual steps to consider. Before upgrading, Apache mod-php needs to be disabled or removed. After upgrading, php-fpm needs to be enabled. For details please check the following remarks according to your operating system of choice.

Debian, Ubuntu, Univention
~~~~~~~~~~~~~~~~~~~~~~~~~~

Before upgrading
^^^^^^^^^^^^^^^^

- Disable or remove any PHP module in Apache2

After upgrading
^^^^^^^^^^^^^^^

- Enable PHP-FPM Apache2 configuration (e.g. `a2enconf php7.3-fpm`, `a2enconf php7.4-fpm` or `a2enconf php8.1-fpm`)

- Restart Apache2 and PHP-FPM

.. code-block:: bareosconfig

   systemctl restart php8.1-fpm

>    systemctl restart apache2

Reviewer 2: You can do both restart on the same line.

> .. code-block:: bareosconfig
> 
>    systemctl restart php8.1-fpm
>    systemctl restart apache2

Reviewer 3: ```suggestion
.. code-block:: shell-session

   root@host:~# systemctl restart php8.1-fpm.service apache2.service
```

This isn't a bareosconfig, but a shell command or shell session.
Also I think it is recommended to write the service type for systemd.

Author: Applied, see https://github.com/bareos/bareos/pull/1287/commits/feacf495ddc53db0df4976c7ace0da87d841150e .

SUSE Linux Enterprise Server (SLES), openSUSE
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Before upgrading
^^^^^^^^^^^^^^^^

- Disable or remove any PHP module in Apache2

After upgrading
^^^^^^^^^^^^^^^

- Ensure a php.ini file is in place (e.g. `/etc/php7/fpm/php.ini` or `/etc/php8/fpm/php.ini`)

- Configure PHP-FPM to your needs (e.g. `/etc/php7/fpm/php-fpm.conf` and `/etc/php7/fpm/php-fpm.d/www.conf`)

- Configure mod_fcgid to your needs `/etc/apache2/conf.d/mod_fcgid.conf`

A minimal example may look like following.

.. code-block:: bareosconfig
   :caption: /etc/apache2/conf.d/mod_fcgid.conf

   <IfModule fcgid_module>
        DirectoryIndex index.php
        <FilesMatch "\.php$">
                SetHandler "proxy:fcgi://127.0.0.1:9000/"
                #CGIPassAuth on
        </FilesMatch>
   </IfModule>

- Restart Apache2 and PHP-FPM

.. code-block:: bareosconfig

>    systemctl restart apache2
>    systemctl restart php-fpm

Reviewer 2: ```suggestion
   systemctl restart php-fpm apache2 
```

Author: Applied, see https://github.com/bareos/bareos/pull/1287/commits/feacf495ddc53db0df4976c7ace0da87d841150e .

   We do provide a specific JobId in the *JobId* command argument in this example. Only the *JobId* given by the placeholder %i will be computed into the cache.

> Upgrade from 21.x.x to 22.x.x

Reviewer: ```suggestion
Upgrade from Bareos 21 or lower to Bareos 22
```

Author: Applied, please see https://github.com/bareos/bareos/pull/1287/commits/da48955442b440c0892f8862c65fa795c4a23872 .

-----------------------------

> |webui| now requires PHP-FPM instead of Apache2 mod-php. Due to this change there are some manual steps to consider
> before and after upgrading. Please have a look at the following remarks according to your operating system of choice.

Reviewer: ```suggestion
|webui| now requires **php-fpm** instead of apache2 **mod-php**. Due to this change there are some manual steps to consider. Before upgrading, apache mod-php needs to be disabled or removed. After upgrading, php-fpm needs to be enabled. For details please check he following remarks according to your operating system of choice.
```

Author: Applied, please see https://github.com/bareos/bareos/pull/1287/commits/da48955442b440c0892f8862c65fa795c4a23872 .
